lib/vyasa_web/components/contexts/read.ex

```diff
@@ -51,26 +51,154 @@ defmodule VyasaWeb.Context.Read do
           }
         } = socket
       ) do
+    send(self(), %{
+      process: MediaBridge,
+      event: :ack_handshake,
+      voice: fn -> Medium.get_voice(src_id, c_no, @default_voice_lang) end,
+      origin: __MODULE__
+    })
+
+    {:ok, socket}
+  end
 
-        send(self(), %{process: MediaBridge, event: :ack_handshake,  voice:  fn -> Medium.get_voice(src_id, c_no, @default_voice_lang) end, origin: __MODULE__})
+  # @bala @ritesh this msg recipient is for the flat version
+  @impl true
+  # received updates from parent liveview when a handshake is init with sesion, does a pub for the voice to use
+  def update(
+        %{
+          id: "read",
+          event: :set_cursor_in_tracklist,
+          tracklist_cursor: tracklist_cursor,
+          track_id: track_id,
+          tracklist_id: tracklist_id
+          # verse_id: verse_id,
+          # chapter_no: chapter_no,
+          # source_id: source_id,
+          # source: source
+        },
+        %{
+          assigns: %{
+            content_action: :show_tracks,
+            tracklist_cursor: curr_cursor
+            # streams: %{tracks: tracks}
+            # tracklist_id: curr_tracklist_id
+            # chap: %Chapter{no: _c_no, source_id: _src_id}
+          }
+        } = socket
+      ) do
+    ## TODO: this is where i want to do an add class where i can target a particular dom node via its id and apply a style class to it
+    ## for this, i want to use the JS.add_class but dont' konw how
+
+    # old_selected = Enum.fil()
+    # current_selected_track = Enum.find()
+    curr_tracklist =
+      Vyasa.Bhaj.get_tracklist(tracklist_id)
+      |> Vyasa.Repo.preload(tracks: [event: [verse: [:source, :chapter]]])
+
+    tracks = curr_tracklist.tracks
+    old_selector_id = "foo_track_" <> Enum.find(tracks, fn t -> t.order === curr_cursor end).id
+    selector_id = "foo_track_" <> track_id
+    class_name = "emphasized-verse"
```

Comment:
currently thhere's a lack of coherence in the navigation targets. 


this is the current integration task that is top priority. 



lack of coherence: 
1. the selector is dependent on the current content_action within the read mode

2. the two functions (one for :show_tracks and another for :show_verses) need to be wired up together. 


^ this is the pending integration work. 

once this is done, the navigation and everything will work

the emphasis and such already works